Rakefile

```diff
@@ -34,3 +36,17 @@ end
 task :release do
   system 'npm publish'
 end
+
+desc 'Update the GitHub pages site'
+task :'gh-pages' do
+  system 'mv -f assets tmp'
+  system 'git checkout origin/gh-pages'
+  system 'mv -f preview.html index.html'
+  system 'rm -rf assets'
+  system 'mv -f tmp assets'
+  system 'git add index.html assets/fonts assets/stylesheets'
+  system "git commit -m 'GitHub pages build #{Time.now}'"
+  system 'git push origin HEAD:gh-pages'
```

Comment:
I recommend moving all of this into a script in like `ci/gh-pages.sh`, then calling from here (or just calling directly from .travis.yml directly).  This way you can tweak it easily and also add stuff like `set -e` if you need it.